### DISCUSSION TITLE
Intended Use Cases for Plane CLI

### DISCUSSION THREAD

**Discussion by akihiro-suzuk1** [jamsocket/plane#807] [Q&A] [answered]
I hope this message finds you well.

I'm currently exploring the Plane platform and have been utilizing the CLI tool provided. While I find it to be a convenient and powerful tool, I would like to gain a deeper understanding of the intended use cases for the Plane CLI.

Could you please provide some insights into the specific scenarios or tasks where the CLI is particularly beneficial? Additionally, I am curious to know if there are any operations that are exclusively or more effectively performed through the CLI compared to other methods, such as using the Plane API directly.

Your guidance will be greatly appreciated as I continue to integrate Plane into our development and operational workflows.

Thank you for your support and for developing such an innovative platform.

Best regards,

**Comment by paulgb** [comment, marked as answer]
Hi @akihiro-suzuk1,

The Plane CLI is essentially just a wrapper around the HTTP API, so there's nothing that can be done through the API. The CLI is meant to be useful when setting up a cluster (e.g. to test spawning and terminating), but there's nothing you can do with it that you couldn't do with a `curl` command, it's just a nicer interface.

I mainly use the CLI for testing things when developing Plane, or when setting up a Plane cluster. I don't use it much for the day-to-day operation of our Plane clusters.

I should mention that there is a second CLI in Plane, [db-cli](https://github.com/jamsocket/plane/blob/main/plane/src/bin/db-cli.rs), in case that's the one you are referring to. It provides some functionality through the database that may not be exposed through the regular CLI, but mostly for development of Plane itself. I don't consider it to be a necessary tool for running a Plane cluster, and it may be removed or folded into the Plane API at some point.

**Reply by akihiro-suzuk1** [reply]
@paulgb 
Thank you very much for the detailed explanation. This clarifies everything I needed to know about the Plane CLI and its use cases. I appreciate your time and insights, and I feel confident moving forward with this understanding.

Best regards,
